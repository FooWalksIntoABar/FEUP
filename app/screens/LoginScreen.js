import React, { Component } from 'react'
import { StatusBar, Text, View } from 'react-native'
import { Container, Content, Form, Item, Input, Label, Button } from 'native-base'

import styles from '../styles/ParentStyles/RegisterScreen.style'
import PropTypes from 'prop-types'

export default class LoginScreen extends Component {
  constructor (props) {
    super(props)
    this.state = {
      email: '',
      emailError: null,
      password: '',
      passwordError: null,
      emailHadInteraction: false,
      passwordHadInteraction: false
    }
  }

  render () {
    return (
      <Container style={styles.registerContainter}>
        <StatusBar hidden />
        <Content contentContainerStyle={styles.contentContainter}>
          <Text style={styles.registerTitle}>Início de sessão</Text>
          <Form>
            <Item floatingLabel error={this.state.emailError} success={!this.state.emailError && this.state.emailHadInteraction} style={styles.inputContainer}>
              <Label style={styles.labelText}>E-mail</Label>
              <Input
                className='email'
                style={styles.labelText}
              />
            </Item>

            <Item floatingLabel error={this.state.passwordError} success={!this.state.passwordError && this.state.passwordHadInteraction} style={styles.inputContainer}>
              <Label style={styles.labelText}>Password</Label>
              <Input
                className='password'
                style={styles.labelText}
                secureTextEntry
              />
            </Item>

            <Button rounded block primary style={styles.submitButton}>
              <Text style={styles.buttonText}>Login</Text>
            </Button>
          </Form>

          <View style={styles.loginTextContainer}>
            <Text style={styles.loginText}>Não possui uma conta?</Text>
            <Text style={styles.loginRegisterText} onPress={() => this.props.navigation.navigate('RegisterMenu')} >Registe-se aqui</Text>
          </View>
        </Content>
      </Container>
    )
  }
}
LoginScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
