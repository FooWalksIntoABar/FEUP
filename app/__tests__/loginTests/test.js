import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import LoginScreen from '../../screens/LoginScreen'

configure({ adapter: new Adapter() })

describe('LoginScreen snapshot', () => {
  it('renders LoginScreen correctly', async () => {
    const wrapper = shallow(<LoginScreen
      navigation={{ navigate: jest.fn() }}
    />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })
})
