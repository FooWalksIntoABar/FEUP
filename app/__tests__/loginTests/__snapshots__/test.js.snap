// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen snapshot renders LoginScreen correctly 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#3db3e2",
    }
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "left": "10%",
        "width": "80%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "LinotteBold",
          "fontSize": 30,
          "marginBottom": "5%",
        }
      }
    >
      Início de sessão
    </Text>
    <Styled(Form)>
      <Styled(Item)
        error={null}
        floatingLabel={true}
        style={
          Object {
            "marginBottom": "5%",
            "marginRight": "5%",
          }
        }
        success={false}
      >
        <Styled(Label)
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        >
          E-mail
        </Styled(Label)>
        <Styled(Input)
          className="email"
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        />
      </Styled(Item)>
      <Styled(Item)
        error={null}
        floatingLabel={true}
        style={
          Object {
            "marginBottom": "5%",
            "marginRight": "5%",
          }
        }
        success={false}
      >
        <Styled(Label)
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        >
          Password
        </Styled(Label)>
        <Styled(Input)
          className="password"
          secureTextEntry={true}
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        />
      </Styled(Item)>
      <Styled(Button)
        block={true}
        primary={true}
        rounded={true}
        style={
          Object {
            "marginTop": "10%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        >
          Login
        </Text>
      </Styled(Button)>
    </Styled(Form)>
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": "15%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "LinotteBold",
            "fontSize": 15,
            "marginBottom": "2%",
          }
        }
      >
        Não possui uma conta?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "LinotteBold",
            "fontSize": 15,
            "marginBottom": "2%",
            "textDecorationLine": "underline",
          }
        }
      >
        Registe-se aqui
      </Text>
    </Component>
  </Styled(Content)>
</Styled(Container)>
`;
